### PR TITLE
feat: usergrant

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: internal/api/authz/permissions_test.go [hunk 1]
```diff
@@ -239,30 +243,36 @@ func Test_MapGrantsToPermissions(t *testing.T) {
 					},
 				},
 			},
-			result: []string{"project.read", "project.read:1"},
+			reuqestPerms: []string{"project.read", "project.read:1"},
```
Comment (adlerhurst) [inline internal/api/authz/permissions_test.go:246]:
```suggestion
			requestPerms: []string{"project.read", "project.read:1"},
```

File: internal/api/authz/permissions_test.go [hunk 2]
```diff
@@ -239,30 +243,36 @@ func Test_MapGrantsToPermissions(t *testing.T) {
...
+			allPerms:     []string{"org.read", "project.read", "project.read:1"},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			result := mapGrantToPermissions(tt.args.requiredPerm, tt.args.grant, tt.args.authConfig)
-			if !equalStringArray(result, tt.result) {
-				t.Errorf("got wrong result, expecting: %v, actual: %v ", tt.result, result)
+			requestPerms, allPerms := mapGrantToPermissions(tt.args.requiredPerm, tt.args.grant, tt.args.authConfig)
+			if !equalStringArray(requestPerms, tt.reuqestPerms) {
```
Comment (adlerhurst) [inline internal/api/authz/permissions_test.go:253]:
```suggestion
			if !equalStringArray(requestPerms, tt.requestPerms) {
```

File: internal/management/repository/eventsourcing/eventstore/project.go [hunk 3]
```diff
@@ -82,15 +82,38 @@ func (repo *ProjectRepo) ReactivateProject(ctx context.Context, id string) (*pro
...
+				}
+				if err == nil {
+					result.Sequence = sequence.CurrentSequence
+					result.Timestamp = sequence.CurrentTimestamp
+				}
+				return result, nil
+			}
+		} else {
+			request.Queries = append(request.Queries, &proj_model.ProjectViewSearchQuery{Key: proj_model.ProjectViewSearchKeyProjectID, Method: global_model.SearchMethodIsOneOf, Value: ids})
+		}
```
Comment (adlerhurst) [inline internal/management/repository/eventsourcing/eventstore/project.go:114]: maybe it makes sense to create a separate func for this

File: internal/management/repository/eventsourcing/eventstore/project.go [hunk 4]
```diff
@@ -348,6 +371,57 @@ func (repo *ProjectRepo) SearchProjectGrants(ctx context.Context, request *proj_
...
+					result.Timestamp = sequence.CurrentTimestamp
+				}
+				return result, nil
+			}
+		} else {
+			request.Queries = append(request.Queries, &proj_model.ProjectGrantViewSearchQuery{Key: proj_model.GrantedProjectSearchKeyGrantID, Method: global_model.SearchMethodIsOneOf, Value: ids})
+		}
+	}
+
+	projects, count, err := repo.View.SearchProjectGrants(request)
```
Comment (adlerhurst) [inline internal/management/repository/eventsourcing/eventstore/project.go:408]:
```suggestion
	grants, count, err := repo.View.SearchProjectGrants(request)
```

File: internal/management/repository/eventsourcing/eventstore/user_grant.go [hunk 5]
```diff
@@ -75,3 +167,33 @@ func (repo *UserGrantRepo) SearchUserGrants(ctx context.Context, request *grant_
...
+			if containsID {
+				return nil
+			}
+		}
+		containsID = listContainsID(ids, projectID)
+		if !containsID {
+			return caos_errors.ThrowPermissionDenied(nil, "EVENT-Shu7e", "Errors.UserGrant.NoPermissionForProject")
+		}
+	}
+	return nil
```
Comment (adlerhurst) [inline internal/management/repository/eventsourcing/eventstore/user_grant.go:187]:
```suggestion
	if authz.HasGlobalPermission(permissions) {
		return nil
	}
	ids := authz.GetAllPermissionCtxIDs(permissions)
	containsID := false
	if grantID != "" {
		containsID = listContainsID(ids, grantID)
		if containsID {
			return nil
		}
	}
	containsID = listContainsID(ids, projectID)
	if !containsID {
		return caos_errors.ThrowPermissionDenied(nil, "EVENT-Shu7e", "Errors.UserGrant.NoPermissionForProject")
	}
	return nil
```

File: internal/management/repository/eventsourcing/eventstore/project.go (hunk 3 above)
Comment (fgerschwiler) [reply internal/management/repository/eventsourcing/eventstore/project.go:114]: yes

**Pull request merged**